core/corehttp/routing.go

```diff
@@ -96,6 +99,60 @@ func (r *contentRouter) PutIPNS(ctx context.Context, name ipns.Name, record *ipn
 	return r.n.Routing.PutValue(ctx, string(name.RoutingKey()), raw)
 }
 
+func (r *contentRouter) GetClosestPeers(ctx context.Context, key cid.Cid) (iter.ResultIter[*types.PeerRecord], error) {
+	// Per the spec, if the peer ID is empty, we should use self.
+	if key == cid.Undef {
+		key = peer.ToCid(r.n.Identity)
+	}
```

Comment:
I wasn't able to find a reference to this in the specs.

If you have access to an endpoint, participating in the Amino DHT (as server or just client), why would users of that endpoint need to know the closest peers to this endpoint? We might as well return an error if an empty CID is supplied?